request_test.go

```diff
@@ -61,7 +62,7 @@ func TestMethods(t *testing.T) {
 
 	for _, row := range table {
 		url := fmt.Sprintf("%s/"+row.name, server.URL)
-		r := New(url)
+		r := New(context.TODO(), url)
```

Comment:
I see that you used context.TODO(). Please change them with context. Background(). I know that in codebase they are exactly same however the intent of the usage is different. We only have one context, no confusion and we use this for tests. So context.Background() is the suitable one.

For more explanation: https://www.programming-books.io/essential/go/context-todo-vs-context-background-d5224e27ff724a33a79cb4e03a5eb333

Reply:
Fixed in https://github.com/binalyze/httpreq/pull/12/commits/afbc08a0ae51ce5a56639f7906ecfc65e0f18886.